### ISSUE TITLE
Questions about the calculation of evaluation

### ISSUE THREAD

**Issue by Avalon-dev** [Jyonn/Legommenders#19]
```
from sklearn.metrics import log_loss, roc_auc_score, ndcg_score, label_ranking_average_precision_score, f1_score

class MRR(Metric):
    name = 'MRR'
    group = True
    minimize = False

def calculate(self, scores: list, labels: list):
   return label_ranking_average_precision_score([labels], [scores])``
```

MRR does not seem to be calculated in this way, can we discuss it？
![image](https://github.com/user-attachments/assets/8c18cbbb-cbc1-41bb-9fec-09d76ad45d12)

**Comment by Jyonn** [comment]
I am willing to hear of that. How to implement MRR in your perspective?

**Comment by Avalon-dev** [comment]
for example
scores = [0.9, 0.8, 0.6, 0.4]
labels = [1, 0, 0, 1]
The result is 0.75.But according to my understanding, MRR should be (1+0.25)/4 = 0.3125. The two results are different. Did I understand something wrong?

**Comment by Jyonn** [comment]
Thanks for your pointing out. We acknowledge that our implementation is not MRR, but LRAP, another metric that is useful in evaluating queries with multiple relevant items.

Here, MRR score is 1.0, which only focus on the first relevant item (rank 1 -> score 1/1=1). The "mean" represents the average score of all queries.

We will revise the implementation as soon as possible. Thanks again for your attention to our Legommenders!

**Comment by Avalon-dev** [comment]
Thank you for your reply. I still have a question. In the MIND dataset, there are multiple “label-1” in the candidate news corresponding to a user's history (that is, the news clicked as the groundtruth), why only focus on the first relevant item when calculating MRR. For example, the MRR result above I still think is 0.3125.

**Comment by Jyonn** [comment]
That is the drawback of the MRR metric. 

(Based on GPT) MRR measures the average reciprocal rank of the first relevant item for a set of queries. It assumes that there is only one relevant item per query or that you are only interested in the first relevant item.

(Based on GPT) If you want to account for all relevant items (not just the first), MRR is not the most appropriate metric. Instead, you might consider metrics like Mean Average Precision (MAP) or Label Ranking Average Precision (LRAP).

**Comment by Jyonn** [comment]
For example, here are two queries:

scores: [[0.9, 0.8, 0.6, 0.4], [0.9, 0.8, 0.6, 0.4]]
labels: [[1, 0, 0, 1], [0, 1, 0, 0]]

MRR = (1/1 + 1/2) / 2 = 3/4

**Comment by Avalon-dev** [comment]
Thank you！
However, I found the following code in many algorithm repositories, such as NAML, to calculate MRR, and I think they calculated it as 0.3125, not 1. So, again, I'm confused. Could you please help me answer it again？
```
def mrr_score(y_true, y_score):
    order = np.argsort(y_score)[::-1]
    y_true = np.take(y_true, order)
    rr_score = y_true / (np.arange(len(y_true)) + 1)
    return np.sum(rr_score) / np.sum(y_true)
```

**Comment by Jyonn** [comment]
Thanks for your pointing out again. 

Firstly, you are correct that many repos use such function to calculate MRR metric, including NAML and FuxiCTR.

Secondly, based on their code, the MRR score on your example is 0.625. It is (1 + 0.25) / 2, rather than (1 + 0.25) / 4.

You may feel straight that the official MRR is not implemented in this way. https://www.evidentlyai.com/ranking-metrics/mean-reciprocal-rank-mrr

I guess the IR researchers have extended MRR into another version that consider all the relevant items. Therefore, I will rename MRR of current version to `MRR0` and also add extended MRR into our MetricPool, namely `MRR`, to be align with those classic repos.

Legommenders will also warn if `MRR` is used for evaluation and suggest researchers to use `MRR0` instead.

**Comment by Avalon-dev** [comment]
Thanks for your reply, I don't have any questions now. 
Thank you for your time and effort to answer my questions

Best wishes!
